streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimestampedStore.java

```diff
@@ -0,0 +1,838 @@
+                    // String format is happening in wrapping stores. So formatted message is thrown from wrapping stores.
+                    throw new ProcessorStateException("Error while putting key %s value %s into store " + name, e);
+                }
+            }
+        }
+
+        @Override
+        public void putAll(final List<KeyValue<Bytes, byte[]>> entries, final WriteBatch batch) throws RocksDBException {
+            for (final KeyValue<Bytes, byte[]> entry : entries) {
+                Objects.requireNonNull(entry.key, "key cannot be null");
+                if (entry.value == null) {
+                    batch.delete(noTimestampColumnFamily, entry.key.get());
+                    batch.delete(withTimestampColumnFamily, entry.key.get());
+                } else {
+                    batch.delete(noTimestampColumnFamily, entry.key.get());
+                    batch.put(withTimestampColumnFamily, entry.key.get(), entry.value);
+                }
+            }
+        }
+
+        @Override
+        public byte[] get(final byte[] rawKey) throws RocksDBException {
+            final byte[] rawValueWithTimestamp = db.get(withTimestampColumnFamily, rawKey);
+            if (rawValueWithTimestamp != null) {
+                return rawValueWithTimestamp;
+            }
+
+            final byte[] rawValue = db.get(noTimestampColumnFamily, rawKey);
+            if (rawValue != null) {
+                final byte[] rawValueWithUnknownTimestamp = getValueWithUnknownTimestamp(rawValue);
+                // this does only work, because the changelog topic contains correct data already
+                // for other format changes, we cannot take this short cut and can only migrate data
+                // from old to new store on put()
+                put(rawKey, rawValueWithUnknownTimestamp);
+                return rawValueWithUnknownTimestamp;
+            }
+
+            return null;
+        }
+
+        @Override
+        public KeyValueIterator<Bytes, byte[]> range(final Bytes from, final Bytes to) {
+            return new RocksDBDualCFRangeIterator(
+                name,
+                db.newIterator(withTimestampColumnFamily),
+                db.newIterator(noTimestampColumnFamily),
+                from,
+                to);
+        }
+
+        @Override
+        public KeyValueIterator<Bytes, byte[]> all() {
+            final RocksIterator innerIterWithTimestamp = db.newIterator(withTimestampColumnFamily);
+            innerIterWithTimestamp.seekToFirst();
+            final RocksIterator innerIterNoTimestamp = db.newIterator(noTimestampColumnFamily);
+            innerIterNoTimestamp.seekToFirst();
+            return new RocksDBDualCFIterator(name, innerIterWithTimestamp, innerIterNoTimestamp);
+        }
+
+        @Override
+        public long approximateNumEntries() throws RocksDBException {
+            return db.getLongProperty(noTimestampColumnFamily, "rocksdb.estimate-num-keys")
```

Comment:
Which do you think is more accurate, this or `db.getLongProperty("rocksdb.estimate-num-keys");` directly?

Reply:
If CF is not specified, not all CF all accessed by default CF only -- thus, we need to add both.

Reply:
I extended `RocksDBTimestampedStoreTest` accordingly to test `approximateNumEntries()`, too.